src/bytes.rs

```diff
@@ -63,6 +79,16 @@ impl ByteBuf {
             bytes: Vec::with_capacity(cap)
         }
     }
+
+    pub fn as_vec(self) -> Vec<u8> {
+        self.bytes
+    }
+}
```

Comment:
It would be generated if there was `impl From<ByteBuf> for Vec<u8>`, not `impl From<Vec<u8>> for ByteBuf`